### PR TITLE
fix: don't activate when downloading versions from settings

### PR DESCRIPTION
**Pull request by codebytere**
Closes https://github.com/electron/fiddle/issues/1238.

`fiddle-core` was installing versions and activating them regardless of origin - when a version is downloaded from settings it should not be activated.

### CODE REVIEW

File: src/renderer/state.ts [hunk 1]
```diff
@@ -610,6 +613,12 @@ export class AppState {
       return;
     }
 
+    // Download the version without setting it as the current version.
+    if (opts?.noActivate) {
+      await this.installer.ensureDownloaded(version);
```
Comment (dsanders11) [inline src/renderer/state.ts:618]: This skips the options passed to `installer.install`, as well as the log statement on line 613/622. Suggestion: pull out the options into `const options = {...};` and then do `this.installer.ensureDownloaded(version, options);`.

Unrelated to this change, but it looks like there's another bug around showing progress on the spinner when downloading under Settings, it works properly on the version select. 🤔

File: src/renderer/components/settings-electron.tsx [hunk 2]
```diff
@@ -383,7 +383,7 @@ export const ElectronSettings = observer(
           buttonProps.onClick = () => {
             isLocal
               ? appState.removeVersion(ver)
-              : appState.downloadVersion(ver);
+              : appState.downloadVersion(ver, { noActivate: true });
```
Comment (ckerr) [inline src/renderer/components/settings-electron.tsx:386]: _nit_

Negated logic like this is often harder to follow / easier to accidentally get wrong later on.

Would it be possible to make the opts `activate` instead, defaulting to true?

**Review by ckerr — APPROVED** [review]
I do have a nit to pick but the main idea in this PR LGTM

File: src/renderer/state.ts (hunk 1 above)
Comment (codebytere) [reply src/renderer/state.ts:618]: What do you mean here precisely? what options would apply to `ensureDownloaded` in this context?

Comment (dsanders11) [reply src/renderer/state.ts:623]: [These options](https://github.com/electron/fiddle/blob/4beb8aba7deb2dc4f60fe874912fe3db45251765/src/renderer/state.ts#L615-L627). Both code paths should probably be consistent in which mirrors they use, and the `progressCallback` option is definitely needed of #1243 will no longer work.

Comment (codebytere) [reply src/renderer/state.ts:623]: @dsanders11 done!

File: src/renderer/components/settings-electron.tsx [hunk 3]
```diff
@@ -383,7 +383,7 @@ export const ElectronSettings = observer(
           buttonProps.onClick = () => {
             isLocal
               ? appState.removeVersion(ver)
-              : appState.downloadVersion(ver);
+              : appState.downloadVersion(ver, { activate: true });
```
Comment (dsanders11) [inline src/renderer/components/settings-electron.tsx:386]: I think this should be flipped to `false`?

File: src/renderer/state.ts [hunk 4]
```diff
@@ -577,12 +577,18 @@ export class AppState {
   }
 
   /**
-   * Download a version of Electron.
+   * Download a version of Electron and set it as the current
+   * version in use unless otherwise specified.
    *
    * @param {RunnableVersion} ver
+   * @param {Object} opts
+   * @param {Boolean} [options.activate=100] - Whether to set ver as current
```
Comment (dsanders11) [inline src/renderer/state.ts:585]: `options.activate=100` => `opts.activate=true`?

**Review by dsanders11 — APPROVED** [review]
:shipit:

**Pull request merged**